### PR TITLE
INT: Add "Change return type" fix

### PR DESCRIPTION
**Pull request by mchernyavsky**
Relates to https://github.com/intellij-rust/intellij-rust/issues/5027.

Applicable to functions, non-inherited methods, and closures.

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/inspections/typecheck/ChangeReturnTypeFixTest.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.typecheck
+
+import org.rust.ide.inspections.RsInspectionsTestBase
+import org.rust.ide.inspections.RsTypeCheckInspection
+
+class ChangeReturnTypeFixTest : RsInspectionsTestBase(RsTypeCheckInspection::class) {
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/inspections/typecheck/ChangeReturnTypeFixTest.kt]: Could you please add tests with type alias to ensure that we support them

File: src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt [hunk 2]
```diff
@@ -352,11 +352,13 @@ class RsPsiFactory(
...
 
-    fun createFunction(
-        text: String
-    ): RsFunction =
+    fun createFunction(text: String): RsFunction =
         createFromText(text)
-            ?: error("Failed to create function element: text")
+            ?: error("Failed to create function element: $text")
+
+    fun createRetType(ty: Ty): RsRetType =
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt]: Maybe it's better to make this API more generic and use `String` instead of `Ty`?

**Comment by mchernyavsky** [comment]
Fixed.

**Comment by mchernyavsky** [comment]
@Undin 
> Complex returning type

Maybe it's better to fix this in a separate PR? It also can improve error messages and `Type Info` action. I can create an issue.

**Comment by Undin** [comment]
> Maybe it's better to fix this in a separate PR? It also can improve error messages and Type Info action. I can create an issue.

👍

**Comment by mchernyavsky** [comment]
Fixed.

**Comment by mchernyavsky** [comment]
Fixed.

**Pull request merged**